docs/ref/models/fields.txt

```diff
@@ -502,6 +502,15 @@ Any combination of these options will result in an error.
     ``True`` will cause the field to have ``editable=False`` and ``blank=True``
     set.
 
+.. note::
+    When using timezones, storing a ``DateField`` in the database is
+    ambiguous because it's incapable of storing timezone aware information, unlike ``DateTimeField``.
+    Unless you know you will need ``DateField``, consider using ``DateTimeField`` instead
+    and convert to a date format for display only.
```

Comment:
The problem, though, is that using a `DateField` with `auto_now(_add)` is, inherently, _not_ "using the data type that matches the data you're modeling." Because the data represented by `auto_now(_add)` is a moment in time when something occurred (a model instance creation or update), and modeling that data with a `DateField` _is_ inherently ambiguous (because which date that event occurred on depends on the timezone of the observer).

Another way of putting it: yes, datetimes are more complex than dates. But with `auto_now(_add)` you always start with a datetime. So you have to deal with the complexity of datetimes either way. Either you deal with it on the way into the database (which Django does for you implicitly and badly by using `datetime.date.today()`, which uses the OS timezone, ignoring even the `TIME_ZONE` setting), in which case you lose information, or you deal with it on the way out of the database at display time.

This does not apply to use of `DateField` in general, just to use of `DateField` with `auto_now(_add)`. I can't think of any situation where a `DateField(auto_now[_add])` would not be better replaced by a `DateTimeField(auto_now[_add])`.